src/runtime/virtcontainers/sandbox.go

```diff
@@ -1859,23 +1988,45 @@ func (s *Sandbox) updateResources(ctx context.Context) error {
 	if err := s.agent.onlineCPUMem(ctx, 0, false); err != nil {
 		return err
 	}
+
 	return nil
 }
 
-func (s *Sandbox) calculateSandboxMemory() int64 {
+func (s *Sandbox) calculateSandboxMemory() (int64, bool, int64) {
```

Comment:
According to `mkswap(8)`,

>  mkswap refuses areas smaller than 10 pages.

Is this handled by the function below? Might be worth adding more unit tests to ensure the minimum swap size is atleast 1MiB or similar? In fact, it might be worth moving the new logic in this function into a new function to make this simpler?

Reply:
I updated function addSwap to handle the size issue.
```
	// Check the size
	pagesize := os.Getpagesize()
	// mkswap refuses areas smaller than 10 pages.
	size = int64(math.Max(float64(size), float64(pagesize*10)))
	// Swapfile need a page to store the metadata
	size += int64(pagesize)
```